manticore/ethereum/account.py

```diff
@@ -102,7 +102,7 @@ def __getattribute__(self, name):
                 contract_account.add(1000)
 
         """
-        if not name.startswith('_'):
+        if not name.startswith('__'):
```

Comment:
This check is wrong - the `__private` attributes are a syntax sugar for `_CLS_private` where `CLS` is the name of the class where a method that uses `__private` is defined.

```
In [9]: class A:
   ...:     def foo_invoker(self):
   ...:         print("A.foo_invoker")
   ...:         self.__foo()
   ...:
   ...:     def __foo(self):
   ...:         print("A.foo")
   ...:
   ...:     def __getattribute__(self, attr):
   ...:         print(f"Getting {attr}")
   ...:         return object.__getattribute__(self, attr)
   ...:

In [10]: A().foo_invoker()
Getting foo_invoker
A.foo_invoker
Getting _A__foo
A.foo
```

Reply:
On the other hand, this would still need to check for all protected attributes (the one starting with `_`) that the EVMAccount/EVMContract defines.

I've proposed a better fix for that issue in #1306.